package/kernelctl/kernelctl

```diff
@@ -0,0 +1,219 @@
+			If you really want to back it up again please run
+
+			$(basename "$0") delete $kernel_name
+			$(basename "$0") backup $kernel_name
+		EOM
+        echo -e "$msg"
+        exit 1
+    else
+        tar cpjf "$kernel_file" lib/modules/* boot/zImage* boot/*.dtb
+        rm -f "$kernelctl_dir"/current.tar.bz2
+        ln "$kernel_file" "$kernelctl_dir"/current.tar.bz2
+    fi
+}
+
+# get available kernels
+get_kernel_names() {
+    mapfile -t kernel_names < <(find "$kernelctl_dir" -path "*.tar.bz2" ! -name current.tar.bz2 -print0 | xargs -0 -I"{}" basename {} .tar.bz2)
+}
+
+# get current kernel
+get_current_kernel_name() {
+    current_kernel_name=$(find "$kernelctl_dir" -samefile "$kernelctl_dir"/current.tar.bz2 ! -name current.tar.bz2 -print0 | xargs -0 -I"{}" basename {} .tar.bz2)
+}
+
+# translate input into a kernel name
+to_kernel_name() {
+    local ker
+    get_kernel_names
+    if [[ "$1" =~ ^[0-9]+$ ]] && ((0 < $1 && $1 <= ${#kernel_names[@]})); then
+        echo "${kernel_names[$(($1 - 1))]}"
+        return
+    elif [[ "$1" = "default" ]]; then
+        ker="vanilla-$(< /etc/version)"
+    else
+        ker="$1"
+    fi
+    if [[ $(echo "${kernel_names[@]}" | grep -ow "$ker" | wc -w) -gt 0 ]]; then
+        echo "$ker"
+    else
+        echo "Can't find $ker in the staging area."
+        exit 1
+    fi
+}
+
+# list available kernels
+list() {
+    get_kernel_names
+    get_current_kernel_name
+    echo -e "${gr}Available kernels:${sf}"
+    for i in "${!kernel_names[@]}"; do
+        if [[ "$current_kernel_name" = "${kernel_names[$i]}" ]]; then cur=" ${bl}*${sf}"; else cur=""; fi
+        echo -e "  ${bf}[$((i + 1))]${sf}\t${kernel_names[$i]}$cur"
+    done
+}
+
+# show the current configured kernel
+show() {
+    get_current_kernel_name
+    echo -e "${gr}Current kernel:${sf}"
+    echo -e "  ${bf}${current_kernel_name}${sf}"
+}
+
+# actually switch kernels
+switch() {
+    tar tjf "$1" | sort -r | xargs -r -I {} bash -c 'if [[ -d "{}" ]]; then rmdir "{}"; else rm "{}"; fi'
+    tar xpjf "$2"
+}
+
+# change the kernel that will boot next time
+set() {
+    new_kernel_name=$(to_kernel_name "$1")
+    new_kernel_file="${kernelctl_dir}/${new_kernel_name}.tar.bz2"
+    current_kernel_file="${kernelctl_dir}/current.tar.bz2"
+    if [[ ! -e "$current_kernel_file" ]]; then
+        read -r -d '' msg <<- EOM
+			${bf}There is no link to the current running kernel in the staging
+			area, you might have accidentally deleted it!${sf}
+			Try making a backup first.
+		EOM
+        echo -e "$msg"
+        exit 1
+    fi
+    if switch "$current_kernel_file" "$new_kernel_file"; then
+        rm -f "$current_kernel_file"
+        ln "$new_kernel_file" "$current_kernel_file"
+        /bin/sync
+        echo "Reboot system to use your newly installed kernel."
+    else
+        read -r -d '' msg <<- EOM
+			${bf}WARNING: failed to extract the new kernel!${sf}
+			Maybe there is not enough space left on the root partition?
```

Comment:
```suggestion
			Make sure there is enough space left in the root partition.
```

Reply:
Space may not be the only issue making this fail, no?